src/main/java/com/openelements/conduct/integration/composite/CompositeCodeOfConductProvider.java

```diff
@@ -0,0 +1,78 @@
+package com.openelements.conduct.integration.composite;
+
+import com.openelements.conduct.data.CodeOfConductProvider;
+import com.openelements.conduct.data.TextfileType;
+import org.jspecify.annotations.NonNull;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import java.util.List;
+import java.util.Objects;
+
+/**
+ * A composite implementation of CodeOfConductProvider that tries multiple providers in order.
+ * This allows for fallback behavior if the primary provider fails.
+ */
+public class CompositeCodeOfConductProvider implements CodeOfConductProvider {
```

Comment:
This change does not make much sense. A project has always exactly 1 Code of Conduct. Therefore you do not want to have multiple providers.